### DISCUSSION TITLE
"Version mismatch!" after update

### DISCUSSION THREAD

**Discussion by JohnKreese** [mbround18/valheim-docker#844] [Help!] [answered]
Is anyone else still getting the "version mismatch!" message after the update today?

I tried doing a docker pull but seems to be latest version already.

I also tried:

`docker exec valheim_container gosu steam bash`

but get error:

`error: failed switching to "steam": operation not permitted`

I only started running this again recently after a few years so have probably forgotten to do something.

Thanks for any help.

Operating System: **Ubuntu 20.04**
Docker Version: **24.0.4**
Docker Compose Version: **1.25.5**
User & Group On Host: **1000:1000**
Are you running with BepInEx? **No**

**Comment by mbround18** [comment]
Just docker exec valheim_container bash, we removed the first user being root

```sh
docker exec valheim_container odin update
```

**Reply by JohnKreese** [reply]
Thanks for that, but I get this message when running `odin update` (also `odin --run-as-root update`):

```
 [ODIN][INFO]  - Checking for updates
 thread 'main' panicked at src/odin/server/process.rs:35:34:
 called `Option::unwrap()` on a `None` value
 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by JohnKreese** [comment]
I changed the "UPDATE_ON_START" value, and that seems to be updating it. I'd still like to know why odin didn't work though.

**Reply by mbround18** [reply, marked as answer]
me too lol ill test it out locally and see if any api has changed

**Comment by jojobyte** [comment]
I just started getting a version mismatch today.

It looks like in spite of `USE_PUBLIC_BETA` being set to `0` or even commented out, the server is updating to the `0.218.16` which is, as of this writing, Public Test, not an official release.

Game client is `0.218.15`

**Reply by mbround18** [reply]
Will investigate
